src/main/java/gregtech/api/recipes/builders/AssemblyLineRecipeBuilder.java

```diff
@@ -0,0 +1,150 @@
+        this.shouldAddResearchRecipe = recipeBuilder.shouldAddResearchRecipe;
+    }
+
+    @Override
+    public AssemblyLineRecipeBuilder copy() {
+        return new AssemblyLineRecipeBuilder(this);
+    }
+
+    @Override
+    public boolean applyPropertyCT(@Nonnull String key, Object value) {
+        if (key.equals(ResearchProperty.KEY)) {
+            this.research(value.toString());
+            return true;
+        }
+        return super.applyPropertyCT(key, value);
+    }
+
+    /**
+     * Do not use! Look for {@link AssemblyLineRecipeBuilder#research(ItemStack, boolean)}
+     */
+    protected AssemblyLineRecipeBuilder research(@Nonnull String researchId) {
+        this.applyProperty(ResearchProperty.getInstance(), researchId);
+        return this;
+    }
+
+    public AssemblyLineRecipeBuilder research(@Nonnull ItemStack researchItem, boolean generateRecipe) {
+        this.shouldAddResearchRecipe = generateRecipe;
+        return research(researchItem);
+    }
+
+    public AssemblyLineRecipeBuilder research(@Nonnull ItemStack researchItem) {
+        return research(researchItem, 144000, GTValues.VA[GTValues.LV]);
+    }
+
+    public AssemblyLineRecipeBuilder research(@Nonnull ItemStack researchItem, int scanDuration, int scanEUt) {
+        return research(researchItem, researchItem.toString(), scanDuration, scanEUt);
+    }
+
+    public AssemblyLineRecipeBuilder research(@Nonnull ItemStack researchItem, @Nonnull String researchId, int scanDuration, int scanEUt) {
+        if (researchItem.isEmpty()) {
+            GTLog.logger.error("Assemblyline Research Items cannot be empty", new IllegalArgumentException());
+            recipeStatus = EnumValidationResult.INVALID;
+        } else {
+            this.researchItem = researchItem;
+            research(researchId);
+        }
+        if (scanDuration <=0) {
+            GTLog.logger.error("Assemblyline Research Duration must be greater than 0", new IllegalArgumentException());
+            recipeStatus = EnumValidationResult.INVALID;
+        } else {
+            this.scanDuration = scanDuration;
+        }
+        if (scanEUt <= 0) {
+            GTLog.logger.error("Assemblyline Research EUt must be greater than 0", new IllegalArgumentException());
+            recipeStatus = EnumValidationResult.INVALID;
+        } else {
+            this.scanEUt = scanEUt;
+        }
+
+        return this;
+    }
+
+    @Override
+    public ValidationResult<Recipe> build() {
+        ValidationResult<Recipe> result = super.build();
+        if (result.getType() == EnumValidationResult.VALID && !this.outputs.isEmpty() && !this.outputs.get(0).isEmpty()) {
+            // only apply the research property if an item was specified
+            if (!this.researchItem.isEmpty() && !recipePropertyStorage.hasRecipeProperty(ResearchProperty.getInstance())) {
+                this.applyProperty(ResearchProperty.getInstance(), this.outputs.get(0).getTranslationKey());
```

Comment:
This should probably just use the itemstack, like the implosion compressor

Reply:
But since the key used is the translation string from the output item of the recipe, we don't really need to worry about data sticks being preserved if the recipe changes, because any recipe that produces the output item will still work.

And if the output item is changed, then that is an entirely new recipe, so then the data stick should be invalidated in that case.

So I don't really see an issue with using an ItemStack, as the itemstack used will always be the recipe output, no matter how the recipe inputs change.